runtime-sdk/modules/evm/src/test.rs

```diff
@@ -280,6 +340,53 @@ impl<C: Config> Runtime for EVMRuntime<C> {
 fn do_test_evm_runtime<C: Config>() {
     let mut mock = mock::Mock::default();
     let mut ctx = mock.create_ctx_for_runtime::<EVMRuntime<C>>(context::Mode::ExecuteTx);
+    let client_keypair =
+        oasis_runtime_sdk::core::common::crypto::mrae::deoxysii::generate_key_pair();
+
+    // This is a macro to avoid mucking with borrow scopes.
+    macro_rules! encode_data {
```

Comment:
These look duplicated from above?

Reply:
It is, but the macro uses idents from the lexical scope (a closure, in some sense) to remove several arguments. Decl macros can't generate macros, so it's copy-pasted. I don't feel bothered by this kind of duplication.

Making a generic function absolutely doesn't work. That's the first thing I tried, but then rustc ends up having trouble determining what kind of `Context` I want.